### PR TITLE
fix(loans): Set min and max exchange rates on runtime upgrade

### PR DESCRIPTION
**Pull request by savudani8**
Adds a migration to set the `min` and `max` exchange rates, if these are zero.

Closes https://github.com/interlay/interbtc/issues/791

### CODE REVIEW

File: .github/workflows/cargo-test.yml [hunk 1]
```diff
@@ -31,6 +31,11 @@ jobs:
         with:
           command: check
           args: --release --workspace
+      - name: try-runtime build
```
Comment (sander2) [inline .github/workflows/cargo-test.yml:34]:
```suggestion
      - name: try-runtime check
```

File: crates/loans/src/lib.rs [hunk 2]
```diff
@@ -343,6 +346,40 @@ pub mod pallet {
         IncentiveReservesReduced(T::AccountId, AssetIdOf<T>, BalanceOf<T>),
     }
 
+    #[pallet::hooks]
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
```
Comment (sander2) [inline crates/loans/src/lib.rs:351]: Please add a guard that checks (and updates) the `palletVersion` so this is only executed once. Note that `palletVersion` is part of Frame - we want to move away from the `StorageVersion` enum that we were using until now

Comment (sander2) [reply crates/loans/src/lib.rs:351]: Hmm yea there is advantage of using the storage version, namely that you can easily see if the migration runs or not. But fair enough about it already been guaranteed to only run once

File: crates/loans/src/lib.rs [hunk 3]
```diff
@@ -343,6 +346,40 @@ pub mod pallet {
...
+            let max_exchange_rate = crate::MaxExchangeRate::<T>::get();
+            if max_exchange_rate.is_zero() {
+                crate::MaxExchangeRate::<T>::put(Rate::from_inner(DEFAULT_MAX_EXCHANGE_RATE));
+            }
+            let min_exchange_rate = crate::MinExchangeRate::<T>::get();
+            if min_exchange_rate.is_zero() {
+                crate::MinExchangeRate::<T>::put(Rate::from_inner(DEFAULT_MIN_EXCHANGE_RATE));
+            }
+            T::DbWeight::get().reads_writes(0, 2)
```
Comment (sander2) [inline crates/loans/src/lib.rs:360]: Nit: should be this, but it doesn't matter a ton
```suggestion
            T::DbWeight::get().reads_writes(2, 2)
```

File: crates/loans/src/lib.rs (hunk 2 above)
Comment (savudani8) [reply crates/loans/src/lib.rs:351]: We can check whether the migration ran if the `MinExchangeRate` and `MaxExchangeRate` storage items are non-zero

File: parachain/runtime/testnet-kintsugi/src/lib.rs [hunk 4]
```diff
@@ -1123,19 +1123,8 @@ pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<Address, RuntimeCall,
...
-    AllPalletsWithSystem,
-    (
-        // "Bound uses of call" <https://github.com/paritytech/substrate/pull/11649>
-        pallet_preimage::migration::v1::Migration<Runtime>,
-        pallet_scheduler::migration::v3::MigrateToV4<Runtime>,
-        pallet_multisig::migrations::v1::MigrateToV1<Runtime>,
-    ),
->;
+pub type Executive =
+    frame_executive::Executive<Runtime, Block, frame_system::ChainContext<Runtime>, Runtime, AllPalletsWithSystem>;
```
Comment (sander2) [inline parachain/runtime/testnet-kintsugi/src/lib.rs:1127]: we should also remove the kintsugi mainnet migrations but we can do that in another pr

**Review by sander2 — APPROVED** [review]
